lib/src/main/kotlin/com/swmansion/starknet/data/types/Responses.kt

```diff
@@ -238,23 +292,119 @@ data class PendingBlockWithTransactionsResponse(
     @SerialName("l1_gas_price")
     override val l1GasPrice: ResourcePrice,
 
+    @SerialName("l1_data_gas_price")
+    override val l1DataGasPrice: ResourcePrice,
+
+    @SerialName("l1_da_mode")
+    override val l1DataAvailabilityMode: L1DAMode,
+
     @SerialName("starknet_version")
     override val starknetVersion: String,
-) : GetBlockWithTransactionsResponse()
+) : GetBlockWithTransactionsResponse(), PendingBlock
+
+@Serializable
+data class TransactionWithReceipt(
+    @Serializable(with = TransactionPolymorphicSerializer::class)
+    @SerialName("transaction")
+    val transaction: Transaction,
+
+    @Serializable(with = TransactionReceiptPolymorphicSerializer::class)
+    @SerialName("receipt")
+    val receipt: TransactionReceipt,
+)
+
+@Serializable
+sealed class GetBlockWithReceiptsResponse : GetBlockWithResponse {
```

Comment:
I think we might want to improve naming for these, seems to me they are becoming less and less readable 😅 

I suggest the `TransactionReceipt` approach:

`GetBlockWith[..]Response` -> `BlockWith[..]`
`BlockWith[..]Response` -> `ProcessedBlockWith[..]`
`PendingBlockWith[..]Response` -> `PendingBlockWith[..]`


where `[...] = [Transactions/TransactionHashes/Receipts]`

This is breaking, but I think it will make interaction with these easier in the future.

WDYT?

Reply:
Sounds good to me

Reply:
Ideally we should just have the same names as in spec.

That being said, I agree that having a `Response` in the class name is abit too verbose, since it's already defined in Responses.kt, but I do wonder if this change won't be too troublesome to users - now, if I'm not mistaken, there will be (eg) GetBlockWithTransactions classes defined in both Responses.kt and Payloads.kt, right? (just to be consistent). Does that mean, that when we want to import both classes, we must import them with an alias? Doesn't that defeat the idea of getting rid of verbose namings?

Reply:
> if I'm not mistaken, there will be (eg) GetBlockWithTransactions classes defined in both Responses.kt and Payloads.kt, right?

No, in `Payloads.kt` the classes are named `GetBlockWith[..]Payload`, so we shouldn't expect any collisions.

Reply:
That is my point - if we want to be consistent, shouldn't we rename these as well?

Reply:
No, not really.

We already use the naming I suggest now in several places, e.g. 
- `GetTransactionStatusPayload` for payload and `TransactionStatus` for response
- `GetTransactionReceiptPayload` for payload and `TransactionReceipt` for response

Reply:
wait, let's take a step back. 

- If I understand correctly, you propose to change the naming for Responses classes - basically ditch the `Response` from the name
- I then ask if we want to be consistent, and drop verbose naming from other classes as well (Payloads); if yes, wouldn't this lead to classes named the same for Payloads and Responses, which would then mean we must import them with aliases
- I get that right now we have *Payload classes for payloads, but by consistent I mean to the Responses approach - why name something a *Payload, if it already is in a Payload.kt, and we do not name classes *Response if they reside in Responses.kt. this seems incosistent.

so my question is - if we want to do it, shoulnd't we do it for Payloads as well? What's with readability then?

Reply:
> If I understand correctly, you propose to change the naming for Responses classes - basically ditch the Response from the name

Not really all of them. I suppose we could also rename `EstimateFeeResponse` -> `EstimateFee` or `FeeEstimate`, but leave classes like `DeployAccountResponse` as is. With logic being that we drop `Response` only if it's not necessary to understand the purpose of the class.

But there's no reason to introduce that many breaking changes at once. Since the spec already introduces changes to the blocks, I thought it would be reasonable to apply changes to them right now, as these are the most problematic with their verbose names. Rest can be done in a separate PR, unless you think we should introduce that change at once.

> if yes, wouldn't this lead to classes named the same for Payloads and Responses, which would then mean we must import them with aliases

This shouldn't really matter, as we use all of the `Payload` classes, at least those located in `Payloads.kt`, only internally. I'm not sure why aren't they `internal`?

> I then ask if we want to be consistent, and drop verbose naming from other classes as well (Payloads);

> I get that right now we have *Payload classes for payloads, but by consistent I mean to the Responses approach - why name something a *Payload, if it already is in a Payload.kt,

I think it introduces unnecessary collision and makes us rely on the source of the import instead of just the name. Besides, the name without `Payload` postfix just doesn't work for `Payload.kt` classes as it doesn't reflect what they are:
- `TransactionReceipt` is a good name for object converted from the response, we don't really need additional `Response` postfix. `TransactionReceipt`, `GetTransactionReceipt`, however, are not good names for the payload with tx hash as only field
- Similarly, `BlockWith[..]` is good name for response, while `BlockWith[..]` as a name for a payload does not reflect what the class is

My intuition was to drop postfix when it's not necessary, not drop it everywhere because the postfix is in the file name, so I don't think we should do it for `Payloads` as it only harms readability.

Reply:
> Rest can be done in a separate PR, unless you think we should introduce that change at once.

Yes, let's do just that to avoid confusion etc. This could be a part of some minor, post 0.7.0 refactor.

> This shouldn't really matter, as we use all of the Payload classes, at least those located in Payloads.kt, only internally. I'm not sure why aren't they internal?

I think they were once public, but am not sure. if that's the case, I'm fine with making them internal and naming them differently to responses.

> I think it introduces unnecessary collision and makes us rely on the source of the import instead of just the name. 

yup, that's my point. let's make sure all payloads are internal, and if they are we can leave them as they are as this won't lead to interface inconsistencies. Otherwise, we should think about a correct approach. sounds good?

Reply:
>  let's make sure all payloads are internal

Should we make the change as a part of this PR or separately?

Reply:
as you prefer

Reply:
#430